src/rest-server/src/templates/dockerContainerScript.mustache

```diff
@@ -23,24 +23,49 @@
 exec 17>/pai/log/DockerContainerDebug.log
 BASH_XTRACEFD=17
 
+RUNTIME_DOCKER_LOG=/pai/log/runtime.docker.pai.log
+RUNTIME_DOCKER_ERR_LOG=/pai/log/runtime.docker.pai.error
+USER_LOG=/pai/log/user.pai.log
+USER_ERROR_LOG=/pai/log/user.pai.error
+DISK_CLEANER_ERROR_FILE=/pai/log/diskCleaner.pai.error
+
+# create runtime docker log here. Since multi shell IO block will write to the files,
+# we just append lines to the log file to avoid flush exits contents.
+touch $RUNTIME_DOCKER_ERR_LOG
+touch $RUNTIME_DOCKER_LOG
+
+# This function is duplicate between dockerContainerScript and yarnContainerScript\
+# Need to remove one if we discard the shell scirpt
+function log_runtime_error()
+{
+  local level=$1
+  local message_type=$2
+  local error_message=$3
+  local timestamp=$(date +%s)
+
+  printf "%s\n" "$timestamp $level $message_type $error_message" >> $RUNTIME_DOCKER_ERR_LOG
+}
+
+
 function exit_handler()
 {
+  rc=$?
```

Comment:
I suggest add sigterm handler as well, to handle yarn's kill

Reply:
fixed